man/systemd-nspawn.xml

```diff
@@ -980,6 +980,20 @@
         effect.</para></listitem>
       </varlistentry>
 
+      <varlistentry>
+        <term><varname>--notify-mode=</varname></term>
+
+        <listitem><para>Configures support for notifications from init process in the container.
+        <varname>--notify-mode=</varname> supports three options: <option>no</option>, <option>yes</option>, and <option>ready</option>.
+        Option <option>no</option> do not enable notifications between systemd-nspawn and
+        init process in the container and systemd-nspawn notifies systemd with a <literal>READY=1</literal>
+        message when the init process is created. Option <option>yes</option> enables notifications
+        but do nothing with them, systemd-nspawn notifies systemd with a <literal>READY=1</literal>
+        message when the init process is created. Option <option>ready</option>
+        enables notifications and wait for the <literal>READY=1</literal> message from
+        the init process in the container before sending its own to systemd.</para></listitem>
+      </varlistentry>
```

Comment:
I agree, this is a better design. So it might better rename the feature from `--notify-mode` to `--wait-ready` or something like that? Which get no parameters it will be on if is set off otherwise and as default nspawn will always listen for notification.